source/d3d12/runtime_d3d12.hpp

```diff
@@ -22,10 +26,23 @@ namespace reshade::d3d12
 
 		bool on_init(const DXGI_SWAP_CHAIN_DESC &desc);
 		void on_reset();
-		void on_present();
+		void on_present(draw_call_tracker& tracker);
+		void copy_depth_stencil(com_ptr<ID3D12Resource> src, com_ptr<ID3D12Resource> dest);
+
 
 		void capture_screenshot(uint8_t *buffer) const override;
 
+#if RESHADE_DX12_CAPTURE_DEPTH_BUFFERS
+		com_ptr<ID3D12Resource> select_depth_texture_save(D3D12_RESOURCE_DESC &texture_desc, const D3D12_HEAP_PROPERTIES *props, D3D12_HEAP_FLAGS *heap_flags);
+#endif
+
+		bool depth_buffer_before_clear = false;
+		bool depth_buffer_more_copies = false;
+		bool extended_depth_buffer_detection = false;
+		unsigned int cleared_depth_buffer_index = 0;
+		int depth_buffer_texture_format = 0; // No depth buffer texture format filter by default
+		unsigned int clear_DSV_iter = 1;
```

Comment:
This should be an atomic. `std::atomic<unsigned int> clear_DSV_iter`

Reply:
OK. Changed